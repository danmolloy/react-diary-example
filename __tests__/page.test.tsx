import '@testing-library/jest-dom'
import { render, screen } from '@testing-library/react'
import Home from '../app/page'
 
describe('Page', () => {
  beforeEach(() => {
    render(<Home />)
  })
  it("home-page is in the document", () => {
    const homePage = screen.getByTestId("home-page")
    expect(homePage).toBeInTheDocument()
  })
  it("React Diary title is in the document", () => {
    const headerText = screen.getByText("React Diary")
    expect(headerText).toBeInTheDocument()
  })
  it("<MonthCalendar /> is in the document with expected events and month", () => {
    const monthCalendar = screen.getByTestId("month-calendar")
    expect(monthCalendar).toBeInTheDocument()
    expect(monthCalendar.textContent).toMatch("October 1986")
    expect(monthCalendar.textContent).toMatch("CLM")
  })
  it("React Diary GitHub link is in the document", () => {
    const githubLink = screen.getByTestId("github-link")
    expect(githubLink.textContent).toMatch("View GitHub")
  })
  it("React Diary npm link is in the document", () => {
    const npmLink = screen.getByTestId("npm-link")
    expect(npmLink.textContent).toMatch("View NPM")
  })
  it("matches snapshot", () => {
    const homePage = screen.getByTestId("home-page")
    expect(homePage).toMatchSnapshot()
  })
})