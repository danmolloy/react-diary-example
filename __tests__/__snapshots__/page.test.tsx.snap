// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page matches snapshot 1`] = `
<main
  class="flex bg-white min-h-screen flex-col items-center justify-between"
  data-testid="home-page"
>
  <div
    class="border-b shadow-sm border-black mt-0 w-full h-12 px-2 flex flex-row items-center"
  >
    <h1
      class="font-semibold text-xl "
    >
      React Diary
    </h1>
  </div>
  <div
    class="p-12 pt-24"
  >
    <table
      class="w-full table-fixed "
      data-testid="month-calendar"
    >
      <thead
        class="w-full h-10"
        data-testid="month-header"
      >
        <tr
          class="w-full "
        >
          <th
            class=""
            colspan="2"
          >
            <button
              class="hover:text-indigo-600 rounded-full hover:bg-slate-100 p-1"
              data-testid="previous-month-button"
              value="Previous Month"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 16 16"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M11.354 1.646a.5.5 0 0 1 0 .708L5.707 8l5.647 5.646a.5.5 0 0 1-.708.708l-6-6a.5.5 0 0 1 0-.708l6-6a.5.5 0 0 1 .708 0"
                  fill-rule="evenodd"
                />
              </svg>
            </button>
          </th>
          <th
            colspan="3"
          >
            <h1>
              October 1986
            </h1>
          </th>
          <th
            colspan="2"
          >
            <button
              class="hover:text-indigo-600 rounded-full text-center hover:bg-slate-100 p-1"
              data-testid="next-month-button"
              value="Next Month"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 16 16"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M4.646 1.646a.5.5 0 0 1 .708 0l6 6a.5.5 0 0 1 0 .708l-6 6a.5.5 0 0 1-.708-.708L10.293 8 4.646 2.354a.5.5 0 0 1 0-.708"
                  fill-rule="evenodd"
                />
              </svg>
            </button>
          </th>
        </tr>
      </thead>
      <thead
        data-testid="days-header"
      >
        <tr>
          <th
            class="border "
          >
            Mon
          </th>
          <th
            class="border "
          >
            Tue
          </th>
          <th
            class="border "
          >
            Wed
          </th>
          <th
            class="border "
          >
            Thu
          </th>
          <th
            class="border "
          >
            Fri
          </th>
          <th
            class="border "
          >
            Sat
          </th>
          <th
            class="border "
          >
            Sun
          </th>
        </tr>
      </thead>
      <tbody
        class=""
      >
        <tr
          data-testid="week-40-row"
        >
          <td
            class="border h-32 w-20"
          >
            <button
              class="bg-slate-50 w-full h-full flex flex-col hover:bg-slate-100 hover:text-indigo-600"
              data-testid="9/29/1986-tile"
            >
              <div
                class="false false p-1 rounded-full w-6 h-6 flex items-center justify-center overflow-visible"
              >
                <h3>
                  29
                </h3>
              </div>
              <ol
                class="text-sm self-center w-[95%] h-full my-1 "
              />
            </button>
          </td>
          <td
            class="border h-32 w-20"
          >
            <button
              class="bg-slate-50 w-full h-full flex flex-col hover:bg-slate-100 hover:text-indigo-600"
              data-testid="9/30/1986-tile"
            >
              <div
                class="false false p-1 rounded-full w-6 h-6 flex items-center justify-center overflow-visible"
              >
                <h3>
                  30
                </h3>
              </div>
              <ol
                class="text-sm self-center w-[95%] h-full my-1 "
              />
            </button>
          </td>
          <td
            class="border h-32 w-20"
          >
            <button
              class="bg-white w-full h-full flex flex-col hover:bg-slate-100 hover:text-indigo-600"
              data-testid="10/1/1986-tile"
            >
              <div
                class="false false p-1 rounded-full w-6 h-6 flex items-center justify-center overflow-visible"
              >
                <h3>
                  1
                </h3>
              </div>
              <ol
                class="text-sm self-center w-[95%] h-full my-1 "
              />
            </button>
          </td>
          <td
            class="border h-32 w-20"
          >
            <button
              class="bg-white w-full h-full flex flex-col hover:bg-slate-100 hover:text-indigo-600"
              data-testid="10/2/1986-tile"
            >
              <div
                class="false false p-1 rounded-full w-6 h-6 flex items-center justify-center overflow-visible"
              >
                <h3>
                  2
                </h3>
              </div>
              <ol
                class="text-sm self-center w-[95%] h-full my-1 "
              />
            </button>
          </td>
          <td
            class="border h-32 w-20"
          >
            <button
              class="bg-white w-full h-full flex flex-col hover:bg-slate-100 hover:text-indigo-600"
              data-testid="10/3/1986-tile"
            >
              <div
                class="false false p-1 rounded-full w-6 h-6 flex items-center justify-center overflow-visible"
              >
                <h3>
                  3
                </h3>
              </div>
              <ol
                class="text-sm self-center w-[95%] h-full my-1 "
              />
            </button>
          </td>
          <td
            class="border h-32 w-20"
          >
            <button
              class="bg-white w-full h-full flex flex-col hover:bg-slate-100 hover:text-indigo-600"
              data-testid="10/4/1986-tile"
            >
              <div
                class="false false p-1 rounded-full w-6 h-6 flex items-center justify-center overflow-visible"
              >
                <h3>
                  4
                </h3>
              </div>
              <ol
                class="text-sm self-center w-[95%] h-full my-1 "
              />
            </button>
          </td>
          <td
            class="border h-32 w-20"
          >
            <button
              class="bg-white w-full h-full flex flex-col hover:bg-slate-100 hover:text-indigo-600"
              data-testid="10/5/1986-tile"
            >
              <div
                class="false false p-1 rounded-full w-6 h-6 flex items-center justify-center overflow-visible"
              >
                <h3>
                  5
                </h3>
              </div>
              <ol
                class="text-sm self-center w-[95%] h-full my-1 "
              />
            </button>
          </td>
        </tr>
        <tr
          data-testid="week-41-row"
        >
          <td
            class="border h-32 w-20"
          >
            <button
              class="bg-white w-full h-full flex flex-col hover:bg-slate-100 hover:text-indigo-600"
              data-testid="10/6/1986-tile"
            >
              <div
                class="false false p-1 rounded-full w-6 h-6 flex items-center justify-center overflow-visible"
              >
                <h3>
                  6
                </h3>
              </div>
              <ol
                class="text-sm self-center w-[95%] h-full my-1 "
              />
            </button>
          </td>
          <td
            class="border h-32 w-20"
          >
            <button
              class="bg-white w-full h-full flex flex-col hover:bg-slate-100 hover:text-indigo-600"
              data-testid="10/7/1986-tile"
            >
              <div
                class="false false p-1 rounded-full w-6 h-6 flex items-center justify-center overflow-visible"
              >
                <h3>
                  7
                </h3>
              </div>
              <ol
                class="text-sm self-center w-[95%] h-full my-1 "
              />
            </button>
          </td>
          <td
            class="border h-32 w-20"
          >
            <button
              class="bg-white w-full h-full flex flex-col hover:bg-slate-100 hover:text-indigo-600"
              data-testid="10/8/1986-tile"
            >
              <div
                class="false false p-1 rounded-full w-6 h-6 flex items-center justify-center overflow-visible"
              >
                <h3>
                  8
                </h3>
              </div>
              <ol
                class="text-sm self-center w-[95%] h-full my-1 "
              />
            </button>
          </td>
          <td
            class="border h-32 w-20"
          >
            <button
              class="bg-white w-full h-full flex flex-col hover:bg-slate-100 hover:text-indigo-600"
              data-testid="10/9/1986-tile"
            >
              <div
                class="false false p-1 rounded-full w-6 h-6 flex items-center justify-center overflow-visible"
              >
                <h3>
                  9
                </h3>
              </div>
              <ol
                class="text-sm self-center w-[95%] h-full my-1 "
              />
            </button>
          </td>
          <td
            class="border h-32 w-20"
          >
            <button
              class="bg-white w-full h-full flex flex-col hover:bg-slate-100 hover:text-indigo-600"
              data-testid="10/10/1986-tile"
            >
              <div
                class="false false p-1 rounded-full w-6 h-6 flex items-center justify-center overflow-visible"
              >
                <h3>
                  10
                </h3>
              </div>
              <ol
                class="text-sm self-center w-[95%] h-full my-1 "
              />
            </button>
          </td>
          <td
            class="border h-32 w-20"
          >
            <button
              class="bg-white w-full h-full flex flex-col hover:bg-slate-100 hover:text-indigo-600"
              data-testid="10/11/1986-tile"
            >
              <div
                class="false false p-1 rounded-full w-6 h-6 flex items-center justify-center overflow-visible"
              >
                <h3>
                  11
                </h3>
              </div>
              <ol
                class="text-sm self-center w-[95%] h-full my-1 "
              />
            </button>
          </td>
          <td
            class="border h-32 w-20"
          >
            <button
              class="bg-white w-full h-full flex flex-col hover:bg-slate-100 hover:text-indigo-600"
              data-testid="10/12/1986-tile"
            >
              <div
                class="false bg-indigo-600 text-white p-1 rounded-full w-6 h-6 flex items-center justify-center overflow-visible"
              >
                <h3>
                  12
                </h3>
              </div>
              <ol
                class="text-sm self-center w-[95%] h-full my-1 "
              />
            </button>
          </td>
        </tr>
        <tr
          data-testid="week-42-row"
        >
          <td
            class="border h-32 w-20"
          >
            <button
              class="bg-white w-full h-full flex flex-col hover:bg-slate-100 hover:text-indigo-600"
              data-testid="10/13/1986-tile"
            >
              <div
                class="false false p-1 rounded-full w-6 h-6 flex items-center justify-center overflow-visible"
              >
                <h3>
                  13
                </h3>
              </div>
              <ol
                class="text-sm self-center w-[95%] h-full my-1 "
              >
                <li
                  class="flex flex-row justify-center items-center flex-nowrap overflow-hidden text-xs mb-1"
                >
                  <div
                    class="h-2 w-2 rounded-full bg-indigo-600   mr-1"
                  />
                  <p
                    class="mr-1 text-gray-500  hidden lg:block"
                  >
                    3:24 AM
                  </p>
                  <p
                    class="text-nowrap hidden sm:block truncate"
                  >
                    CLM
                  </p>
                </li>
              </ol>
            </button>
          </td>
          <td
            class="border h-32 w-20"
          >
            <button
              class="bg-white w-full h-full flex flex-col hover:bg-slate-100 hover:text-indigo-600"
              data-testid="10/14/1986-tile"
            >
              <div
                class="false false p-1 rounded-full w-6 h-6 flex items-center justify-center overflow-visible"
              >
                <h3>
                  14
                </h3>
              </div>
              <ol
                class="text-sm self-center w-[95%] h-full my-1 "
              >
                <li
                  class="flex flex-row justify-center items-center flex-nowrap overflow-hidden text-xs mb-1"
                >
                  <div
                    class="h-2 w-2 rounded-full bg-indigo-600   mr-1"
                  />
                  <p
                    class="mr-1 text-gray-500  hidden lg:block"
                  >
                    12:00 PM
                  </p>
                  <p
                    class="text-nowrap hidden sm:block truncate"
                  >
                    Lunch Meeting
                  </p>
                </li>
              </ol>
            </button>
          </td>
          <td
            class="border h-32 w-20"
          >
            <button
              class="bg-white w-full h-full flex flex-col hover:bg-slate-100 hover:text-indigo-600"
              data-testid="10/15/1986-tile"
            >
              <div
                class="false false p-1 rounded-full w-6 h-6 flex items-center justify-center overflow-visible"
              >
                <h3>
                  15
                </h3>
              </div>
              <ol
                class="text-sm self-center w-[95%] h-full my-1 "
              />
            </button>
          </td>
          <td
            class="border h-32 w-20"
          >
            <button
              class="bg-white w-full h-full flex flex-col hover:bg-slate-100 hover:text-indigo-600"
              data-testid="10/16/1986-tile"
            >
              <div
                class="false false p-1 rounded-full w-6 h-6 flex items-center justify-center overflow-visible"
              >
                <h3>
                  16
                </h3>
              </div>
              <ol
                class="text-sm self-center w-[95%] h-full my-1 "
              />
            </button>
          </td>
          <td
            class="border h-32 w-20"
          >
            <button
              class="bg-white w-full h-full flex flex-col hover:bg-slate-100 hover:text-indigo-600"
              data-testid="10/17/1986-tile"
            >
              <div
                class="false false p-1 rounded-full w-6 h-6 flex items-center justify-center overflow-visible"
              >
                <h3>
                  17
                </h3>
              </div>
              <ol
                class="text-sm self-center w-[95%] h-full my-1 "
              />
            </button>
          </td>
          <td
            class="border h-32 w-20"
          >
            <button
              class="bg-white w-full h-full flex flex-col hover:bg-slate-100 hover:text-indigo-600"
              data-testid="10/18/1986-tile"
            >
              <div
                class="false false p-1 rounded-full w-6 h-6 flex items-center justify-center overflow-visible"
              >
                <h3>
                  18
                </h3>
              </div>
              <ol
                class="text-sm self-center w-[95%] h-full my-1 "
              />
            </button>
          </td>
          <td
            class="border h-32 w-20"
          >
            <button
              class="bg-white w-full h-full flex flex-col hover:bg-slate-100 hover:text-indigo-600"
              data-testid="10/19/1986-tile"
            >
              <div
                class="false false p-1 rounded-full w-6 h-6 flex items-center justify-center overflow-visible"
              >
                <h3>
                  19
                </h3>
              </div>
              <ol
                class="text-sm self-center w-[95%] h-full my-1 "
              />
            </button>
          </td>
        </tr>
        <tr
          data-testid="week-43-row"
        >
          <td
            class="border h-32 w-20"
          >
            <button
              class="bg-white w-full h-full flex flex-col hover:bg-slate-100 hover:text-indigo-600"
              data-testid="10/20/1986-tile"
            >
              <div
                class="false false p-1 rounded-full w-6 h-6 flex items-center justify-center overflow-visible"
              >
                <h3>
                  20
                </h3>
              </div>
              <ol
                class="text-sm self-center w-[95%] h-full my-1 "
              />
            </button>
          </td>
          <td
            class="border h-32 w-20"
          >
            <button
              class="bg-white w-full h-full flex flex-col hover:bg-slate-100 hover:text-indigo-600"
              data-testid="10/21/1986-tile"
            >
              <div
                class="false false p-1 rounded-full w-6 h-6 flex items-center justify-center overflow-visible"
              >
                <h3>
                  21
                </h3>
              </div>
              <ol
                class="text-sm self-center w-[95%] h-full my-1 "
              />
            </button>
          </td>
          <td
            class="border h-32 w-20"
          >
            <button
              class="bg-white w-full h-full flex flex-col hover:bg-slate-100 hover:text-indigo-600"
              data-testid="10/22/1986-tile"
            >
              <div
                class="false false p-1 rounded-full w-6 h-6 flex items-center justify-center overflow-visible"
              >
                <h3>
                  22
                </h3>
              </div>
              <ol
                class="text-sm self-center w-[95%] h-full my-1 "
              />
            </button>
          </td>
          <td
            class="border h-32 w-20"
          >
            <button
              class="bg-white w-full h-full flex flex-col hover:bg-slate-100 hover:text-indigo-600"
              data-testid="10/23/1986-tile"
            >
              <div
                class="false false p-1 rounded-full w-6 h-6 flex items-center justify-center overflow-visible"
              >
                <h3>
                  23
                </h3>
              </div>
              <ol
                class="text-sm self-center w-[95%] h-full my-1 "
              />
            </button>
          </td>
          <td
            class="border h-32 w-20"
          >
            <button
              class="bg-white w-full h-full flex flex-col hover:bg-slate-100 hover:text-indigo-600"
              data-testid="10/24/1986-tile"
            >
              <div
                class="false false p-1 rounded-full w-6 h-6 flex items-center justify-center overflow-visible"
              >
                <h3>
                  24
                </h3>
              </div>
              <ol
                class="text-sm self-center w-[95%] h-full my-1 "
              />
            </button>
          </td>
          <td
            class="border h-32 w-20"
          >
            <button
              class="bg-white w-full h-full flex flex-col hover:bg-slate-100 hover:text-indigo-600"
              data-testid="10/25/1986-tile"
            >
              <div
                class="false false p-1 rounded-full w-6 h-6 flex items-center justify-center overflow-visible"
              >
                <h3>
                  25
                </h3>
              </div>
              <ol
                class="text-sm self-center w-[95%] h-full my-1 "
              />
            </button>
          </td>
          <td
            class="border h-32 w-20"
          >
            <button
              class="bg-white w-full h-full flex flex-col hover:bg-slate-100 hover:text-indigo-600"
              data-testid="10/26/1986-tile"
            >
              <div
                class="false false p-1 rounded-full w-6 h-6 flex items-center justify-center overflow-visible"
              >
                <h3>
                  26
                </h3>
              </div>
              <ol
                class="text-sm self-center w-[95%] h-full my-1 "
              />
            </button>
          </td>
        </tr>
        <tr
          data-testid="week-44-row"
        >
          <td
            class="border h-32 w-20"
          >
            <button
              class="bg-white w-full h-full flex flex-col hover:bg-slate-100 hover:text-indigo-600"
              data-testid="10/27/1986-tile"
            >
              <div
                class="false false p-1 rounded-full w-6 h-6 flex items-center justify-center overflow-visible"
              >
                <h3>
                  27
                </h3>
              </div>
              <ol
                class="text-sm self-center w-[95%] h-full my-1 "
              />
            </button>
          </td>
          <td
            class="border h-32 w-20"
          >
            <button
              class="bg-white w-full h-full flex flex-col hover:bg-slate-100 hover:text-indigo-600"
              data-testid="10/28/1986-tile"
            >
              <div
                class="false false p-1 rounded-full w-6 h-6 flex items-center justify-center overflow-visible"
              >
                <h3>
                  28
                </h3>
              </div>
              <ol
                class="text-sm self-center w-[95%] h-full my-1 "
              />
            </button>
          </td>
          <td
            class="border h-32 w-20"
          >
            <button
              class="bg-white w-full h-full flex flex-col hover:bg-slate-100 hover:text-indigo-600"
              data-testid="10/29/1986-tile"
            >
              <div
                class="false false p-1 rounded-full w-6 h-6 flex items-center justify-center overflow-visible"
              >
                <h3>
                  29
                </h3>
              </div>
              <ol
                class="text-sm self-center w-[95%] h-full my-1 "
              />
            </button>
          </td>
          <td
            class="border h-32 w-20"
          >
            <button
              class="bg-white w-full h-full flex flex-col hover:bg-slate-100 hover:text-indigo-600"
              data-testid="10/30/1986-tile"
            >
              <div
                class="false false p-1 rounded-full w-6 h-6 flex items-center justify-center overflow-visible"
              >
                <h3>
                  30
                </h3>
              </div>
              <ol
                class="text-sm self-center w-[95%] h-full my-1 "
              />
            </button>
          </td>
          <td
            class="border h-32 w-20"
          >
            <button
              class="bg-white w-full h-full flex flex-col hover:bg-slate-100 hover:text-indigo-600"
              data-testid="10/31/1986-tile"
            >
              <div
                class="false false p-1 rounded-full w-6 h-6 flex items-center justify-center overflow-visible"
              >
                <h3>
                  31
                </h3>
              </div>
              <ol
                class="text-sm self-center w-[95%] h-full my-1 "
              />
            </button>
          </td>
          <td
            class="border h-32 w-20"
          >
            <button
              class="bg-slate-50 w-full h-full flex flex-col hover:bg-slate-100 hover:text-indigo-600"
              data-testid="11/1/1986-tile"
            >
              <div
                class="false false p-1 rounded-full w-6 h-6 flex items-center justify-center overflow-visible"
              >
                <h3>
                  1
                </h3>
              </div>
              <ol
                class="text-sm self-center w-[95%] h-full my-1 "
              />
            </button>
          </td>
          <td
            class="border h-32 w-20"
          >
            <button
              class="bg-slate-50 w-full h-full flex flex-col hover:bg-slate-100 hover:text-indigo-600"
              data-testid="11/2/1986-tile"
            >
              <div
                class="false false p-1 rounded-full w-6 h-6 flex items-center justify-center overflow-visible"
              >
                <h3>
                  2
                </h3>
              </div>
              <ol
                class="text-sm self-center w-[95%] h-full my-1 "
              />
            </button>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
  <div
    class=" w-full h-24 flex flex-col items-center justify-center bg-slate-100"
  >
    <a
      class="flex flex-row items-center m-1 text-2xl hover:text-indigo-600 hover:underline"
      data-testid="github-link"
      href="https://github.com/danmolloy/react-diary"
      target="_blank"
    >
      <svg
        color="black"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        style="color: black;"
        viewBox="0 0 496 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M165.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6zm-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3zm44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9zM244.8 8C106.1 8 0 113.3 0 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C428.2 457.8 496 362.9 496 252 496 113.3 383.5 8 244.8 8zM97.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1zm-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7zm32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1zm-11.4-14.7c-1.6 1-1.6 3.6 0 5.9 1.6 2.3 4.3 3.3 5.6 2.3 1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2z"
        />
      </svg>
      <p
        class="ml-1 text-base"
      >
        View GitHub
      </p>
    </a>
    <a
      class="flex flex-row items-center m-1 text-2xl hover:text-indigo-600 hover:underline"
      data-testid="npm-link"
      href="https://www.npmjs.com/package/react-diary"
      target="_blank"
    >
      <svg
        color="black"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        style="color: black;"
        viewBox="0 0 576 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M288 288h-32v-64h32v64zm288-128v192H288v32H160v-32H0V160h576zm-416 32H32v128h64v-96h32v96h32V192zm160 0H192v160h64v-32h64V192zm224 0H352v128h64v-96h32v96h32v-96h32v96h32V192z"
        />
      </svg>
      <p
        class="ml-1 text-base"
      >
        View NPM
      </p>
    </a>
  </div>
</main>
`;
